apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c

```diff
@@ -0,0 +1,848 @@
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         audio_pcm_bytes_to_frames_p
+* @brief            convert bytes to frame
+* @scenario         get configuration value of pcm and calculate byte to frame
+* @apicovered       pcm_bytes_to_frames
+* @precondition     pcm should be opened before.
+* @postcondition    NA
+*/
+static void utc_audio_pcm_bytes_to_frames_p(void)
+{
+
+	unsigned int frame_size;
+	frame_size = pcm_bytes_to_frames(g_pcm, g_byte_size);
+	TC_ASSERT_GT("pcm_bytes_to_frame", frame_size, 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         audio_pcm_bytes_to_frames_n
+* @brief            convert bytes to frame
+* @scenario         get configuration value of NULL pcm and calculate byte to frame
+* @apicovered       pcm_bytes_to_frames
+* @precondition     NA
+* @postcondition    NA
+*/
+static void utc_audio_pcm_bytes_to_frames_n(void)
+{
+	unsigned int frame;
+	frame = pcm_frames_to_bytes(NULL, 0);
+	TC_ASSERT_LEQ("pcm_frames_to_bytes", frame, 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         utc_audio_pcm_format_to_bits_p
+* @brief            get bits information based on pcm's format
+* @scenario         get bits of format value of pcm
+* @apicovered       pcm_format_to_bits
+* @precondition     pcm should be opened before.
+* @postcondition    NA
+*/
+static void utc_audio_pcm_format_to_bits_p(void)
+{
+	enum pcm_format format;
+	format = pcm_get_format(g_pcm);
+	TC_ASSERT_NEQ("pcm_format_to_bits", format, PCM_FORMAT_NONE);
+	TC_ASSERT_NEQ("pcm_format_to_bits", pcm_format_to_bits(format), 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         utc_audio_pcm_format_to_bits_n
+* @brief            get bits information based on pcm's format
+* @scenario        get bits of format value with invalid format
+* @apicovered       pcm_format_to_bits
+* @precondition     NA
+* @postcondition    NA
+*/
+static void utc_audio_pcm_format_to_bits_n(void)
+{
+	TC_ASSERT_EQ("pcm_format_to_bits", pcm_format_to_bits(PCM_FORMAT_NONE), 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         audio_pcm_readi_p
+* @brief            read captured data from chip
+* @scenario         record voice and write data in specific file
+* @apicovered       pcm_readi
+* @precondition     pcm should be opened before
+* @postcondition    NA
+*/
+static void utc_audio_pcm_readi_p(void)
+{
+	int fd;
+	char input_str[16];
+	ssize_t size;
+	ssize_t bytes;
+	unsigned int frames_read = 0;
+	unsigned int total_frames_read = 0;
+	unsigned int rate;
+	int capturing;
+	int ret;
+
+	size = pcm_get_buffer_size(g_pcm);
+	rate = pcm_get_rate(g_pcm);
+	bytes = rate * (pcm_format_to_bits(pcm_get_format(g_pcm)) / 8) * pcm_get_channels(g_pcm) * AUDIO_RECORD_DURATION;
+
+	g_record_buffer = (char *)malloc(bytes); //rate * bit * channels * duration
+	TC_ASSERT("pcm_readi_p", g_record_buffer);
+
+	printf("Record will be start for 3s, press any key to start\n");
+	fflush(stdout);
+	if (gets(input_str) == NULL) {
+		TC_ASSERT("audio_pcm_readi", gets(input_str));
```

Comment:
Same

Reply:
Before using tc macro, call free. This can be a leak

Reply:
fixed .